src/Expression.php

```diff
@@ -587,7 +587,7 @@ public function execute(object $connection = null)
                     // workaround to support LOB data type 2/2, see https://github.com/doctrine/dbal/pull/2434
                     $statement->boundValues[$key] = $val;
                     if ($type === \PDO::PARAM_STR) {
-                        $bind = $statement->bindParam($key, $statement->boundValues[$key], $type, strlen((string) $val));
+                        $bind = $statement->bindParam($key, $statement->boundValues[$key], $type, mb_strlen((string) $val));
```

Comment:
Be very carefull here, it may require a binary length here!!

Reply:
yes most likely it should be binary

Reply:
I do not know, should be checked, it is probably for maximal byte length to allocate, my guess.